specs/chugsplash-manager.md

```diff
@@ -0,0 +1,178 @@
+  * The `activeBundleId` MUST be set to `bytes32(0)`.
+  * The ChugSplashManager MUST emit the event ChugSplashBundleCompleted with the active bundle ID, the executor's address, and the action index. The ChugSplashManager MUST announce this event to the registry.
+
+## Cancelling a ChugSplash bundle
+
+* A ChugSplash bundle MUST only be cancellable by the `admin` of the `ChugSplashManager`.
+* Bundles MUST be in the `APPROVED` state before they can be cancelled.
+* Cancelling a bundle MUST put the bundle in the `CANCELLED` state.
+* Cancelling a bundle MUST remove the active bundle.
+* The ChugSplashManager MUST increase the `debt` owed to the current executor by `BUNDLE_EXECUTION_BOND_AMOUNT`.
+* The ChugSplashManager MUST increase the `totalDebt` by the `BUNDLE_EXECUTION_BOND_AMOUNT`.
+* The ChugSplashManager MUST emit the event ChugSplashBundleCancelled with the ID of the bundle and the total number of actions executed. The ChugSplashManager MUST announce this event to the registry.
+
+## Withdrawing ETH
+
+* ETH stored in the contract MUST only be withdrawable by the `admin` of the `ChugSplashManager`.
+* The admin ETH withdrawal function MUST NOT be callable while a bundle is in the `APPROVED` state.
+* The admin ETH withdrawal function MUST only allow the admin to withdraw the balance of the account minus `totalDebt`.
+* The ChugSplashManager MUST emit the ETHWithdrawn event with the recipient and the amount. The ChugSplashManager MUST announce this event to the registry.
+
+## Claiming executor payments
+
+* Executors MUST be able to claim payments at any time.
+* Executors MUST NOT be able to claim more payments than their `debt` amount.
+* The ChugSplashManager MUST update the `debt` and `totalDebt` values to reflect the payment.
+* The ChugSplashManager MUST emit the ExecutorPayment event with the recipient and amount. The ChugSplashManager MUST announce this event to the registry.
+
+## Depositing ETH
+
+* Any address should be able to deposit ETH into the `ChugSplashManager` contract at any time.
+
+## Claiming a bundle
+
+* Anyone should be able to claim a bundle.
+* The executor's `debt` plus the `msg.value` MUST be greater than or equal to the `executorBondAmount`.
+* The bundle being claimed MUST have an `APPROVED` status.
+* The current `block.timestamp` MUST be greater than the `timeClaimed` plus the `executionLockTime`.
+* Claiming a bundle MUST set the `timeClaimed` to be the `block.timestamp`.
+* Claiming a bundle MUST set the `selectedExecutor` to be the executor that claimed the bundle.
+* The ChugSplashManager MUST increase the `debt` owed to the current executor by the `executorBondAmount`.
+* The ChugSplashManager MUST increase the `totalDebt` by the `executorBondAmount`.
+* The ChugSplashManager MUST emit the event ChugSplashBundleClaimed with the claimed bundle ID and the address of the executor. The ChugSplashManager MUST announce this event to the registry.
+
+## Setting the executor's bond amount
```

Comment:
For simplicity for now let's get rid of the ability to set the executor bond and use a reasonable default.